content/blog/posts/2022-07-08-go-serverless-functions-with-ember-and-sst-on-aws.md

```diff
@@ -0,0 +1,200 @@
+
+We'll also want to do this using environment variables so we can later handle the different `ApiEndpoint` URL that will support the deployed production instance of our application. So we'll add a dot env file with `EMBER_APP_API_URL` as an environment variable and set that equal to the `ApiEndpoint`.
+
+```javascript
+// ./application/route.js
+import Route from '@ember/routing/route';
+import ENV from 'frontend/config/environment';
+
+export default class ApplicationRoute extends Route {
+  async model() {
+    const response = await fetch(ENV.APP.EMBER_APP_API_URL);
+
+    const message = await response.json();
+
+    return message;
+  }
+}
+```
+
+We'll also modify the `./config/environment.js` file to set this environment variable in the Ember app for use later on:
+
+```javascript
+// .config/environment.js
+'use strict';
+require('dotenv').config();
+
+module.exports = function (environment) {
+  let ENV = {
+    modulePrefix: 'frontend',
+    environment,
+    rootURL: '/',
+    locationType: 'history',
+    EmberENV: {
+      FEATURES: {},
+      EXTEND_PROTOTYPES: {
+        Date: false,
+      },
+    },
+
+    APP: {},
+  };
+
+  if (environment === 'development') {
+    ENV.APP.EMBER_APP_API_URL = process.env.EMBER_APP_API_URL_DEV;
+  }
+
+  if (environment === 'test') {
+    ENV.locationType = 'none';
+
+    // keep test console output quieter
+    ENV.APP.LOG_ACTIVE_GENERATION = false;
+    ENV.APP.LOG_VIEW_LOOKUPS = false;
+
+    ENV.APP.rootElement = '#ember-testing';
+    ENV.APP.autoboot = false;
+  }
+
+  if (environment === 'production') {
+    ENV.APP.EMBER_APP_API_URL = process.env.EMBER_APP_API_URL_PROD;
+  }
+
+  return ENV;
+};
+```
+
+And finally, we'll tweak the serverless function to respond with JSON rather than a string:
+
+```go
+// ./services/functions/lambda/main.go
+package main
+
+import (
+	"encoding/json"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+func Handler(request events.APIGatewayV2HTTPRequest) (events.APIGatewayProxyResponse, error) {
+	data := map[string]string{"message": "Hello, World!"}
+	json, err := json.Marshal(data)
+
+	return events.APIGatewayProxyResponse{
+		Body:       string(json),
+		StatusCode: 200,
+	}, err
+}
+
+func main() {
+	lambda.Start(Handler)
+}
+```
+
+With that wired up you can fire up the Ember app as you normally would and check in the browser that the serverless function is returning the data we expected:
+
+![Browser showing the Go serverless function response](/img/blog/serverlesss-functions-direct-to-aws-with-ember/browser-showing-go-serverless-response.png)
+
+And there is our Ember app with a Go-based serverless function that is deployable to AWS, and we only need to take one more step to deploy it!
```

Comment:
And there is our Ember app with a Go-based serverless function! We only ~need to take~ have one more step to deploy it!